misk/web/tabs/config/src/components/ConfigComponent.tsx

```diff
@@ -0,0 +1,86 @@
+// tslint:disable-next-line:no-var-requires
+const YAML = require("json2yaml")
+import * as React from "react"
+import styled from "styled-components"
+import { IConfigFile } from "../containers/TabContainer"
+
+interface IConfigProps {
+  files: IConfigFile[]
+  status: string
+}
+
+const Container = styled.div`
+  margin-left: 180px;
+  margin-top: 20px;
+`
+
+export default class ConfigComponent extends React.PureComponent<IConfigProps> {
+  indent(spaces: number) {
+    let final = ""
+    for (let i = 0; i < spaces; i++) {
+      final += "\ "
+    }
+    return final
+  }
+
+  toYaml(json: string) {
+    let final = ""
+    let level = 0
+    for (const c of json) {
+      switch (c) {
+        case "{": 
+          level++
+          break
+        case "}":
+          level--
+          final += "\n" + this.indent(level)
+          break
+        case ",": 
+          final += "\n" + this.indent(level)
+          break
+        case ":":
+          // write parser for string
+          final += ": "
+          break
+        case "\"":
+          break
+        case "\,":
+          break
+        default:
+          final += c
+      }
+    }
+    return(final)
+  }
+
+  oldToYaml(json: string) {
```

Comment:
what’s _old_ ? Could ya add a comment?

Reply:
old is the first iteration I did a few weeks ago that was very rough join/split chaining that worked well enough for demo but isn't a long term solution.